tox.ini

```diff
@@ -127,6 +127,11 @@ extend-ignore =
 # sqlfluff uses flake8-docstrings https://pypi.org/project/flake8-docstrings/
 # this is to assist with the sphinx based autodoc
 docstring-convention = google
+quotes = double
+
+[isort]
+line_length = 127
```

Comment:
Should this be the black default of 88?

```suggestion
line_length = 88
```

Reply:
i've set it to same as set in `flake8` section
we do same in out projects

Reply:
Oh you're right.

So, I'm just confused why it complained about your long line, which was 115 characters long so should have been fine with this config.

@jpy-git / @barrywhart any ideas here?

Reply:
tbh this is unrelated to the PR so shouldn't really be in here.

What is isort and why are we adding it?

Reply:
because when i've ran flake8 lint - it shows up with numerous warnings abt styling which are used as codestyle but not defined in `tox.ini`:
* double quotes
* imports
* etc
![image](https://user-images.githubusercontent.com/34435869/144765596-e62cfa99-07b5-4fe9-8389-a008e6216f9e.png)
if you are suggesting to remove all styling stuff from here - i can do this
in this case changes will not pass linting stuff (flake8)

Reply:
I don’t get those errors when running flake8 or black. So you have some sort of global default which prefers single quotes or something?

Reply:
No worries, it's just useful to have context when these changes that affect larger areas of the project.

The double quotes one is weird as its referring to lines that your PR hasn't even touched and we use double quotes everywhere. Let me take a look to see if I have these issues.

Reply:
most of project is about doublequoted string literals - however this approach is outdated and not encouraged by nowadays python styling
But anyway - this is a matter of taste/style your project prefer
So you can choose any style - but it should be global across the project

Reply:
I have also checked and as @tunetheweb rightly says linting is working fine without these changes, pretty sure you've have some global settings enabled outside of this project. Please remove these config changes.

Reply:
have you tried to run flake8 locally using root of the project as working dir?
it is generated numerous codestyle issues as stated on screen above

Reply:
- Do you have a flake8 config (or similar) in ~/.config ?
- are you using a virtual environment 
```
tox -e py --devenv .venv
source .venv/bin/activate
```
- are you running flake8 via `tox -e linting`

Reply:
> have you tried to run flake8 locally using root of the project as working dir?
> it is generated numerous codestyle issues as stated on screen above

Yes, works fine

Reply:
> 

and no warnings there?

Reply:
> Do you have a flake8 config (or similar) in ~/.config ?
> are you using a virtual environment
no i don't use venvs
```bash
dmytro.kulyk@MB-DAT-564087 datalake % ls -lah ~/.config
total 0
drwxr-xr-x   6 dmytro.kulyk  staff   192B Nov 17 17:01 .
drwxr-xr-x+ 82 dmytro.kulyk  staff   2.6K Dec  5 23:19 ..
drwxr-xr-x   2 dmytro.kulyk  staff    64B Oct  9 13:39 azure-datalake-store
drwx------   3 dmytro.kulyk  staff    96B Aug 25 16:29 configstore
drwxr-xr-x   3 dmytro.kulyk  staff    96B Mar  4  2021 jgit
drwxr-xr-x   3 dmytro.kulyk  staff    96B Nov 17 17:01 k9s
```

Reply:
That is correct. Many people have contributed here and this is the first we’ve heard of this problem. It really seems like you have some defaults (using single quotes as you seem to prefer).

I’m not against adding above config, to help out other contributors that also have different local settings, but so agree it should be a separate PR so we can discuss it’s merits or not.

Reply:
I'm pretty sure you're not running in a virtual environment as a quick google search reveals Q000 is from `flake8-quotes` and C812 is from `flake8-commas` which are plugins we don't use in this project.

Please create and activate a virtual environment as described above and in CONTRIBUTING.md

Reply:
if you still have the issues then run
`pip freeze`
and 
`which python`
and send me the output

Reply:
@KulykDmytro okay then use a venv

The point of a virtual environment is everyone uses the same setup. That way we don't deal with issue of people having different installs/configs and its common practice.

Reply:
Also note that Black specifically requires double quotes and will modify code so that it complies. We **definitely** don't want to configure `flake8` to enforce the opposite setting, nor change existing code to use single quotes.

https://black.readthedocs.io/en/stable/the_black_code_style/current_style.html#strings

Since we've standardized on Black double quotes, we should stay with it.